### PR TITLE
Compas main

### PR DESCRIPTION
**Pull request by chenkasirer**
Updated the `Data` interface in `Beam` and `Assembly` to comply with the changes in COMPAS core intended to be released in version 2.0.

### What type of change is this?

- [ ] Bug fix in a **backwards-compatible** manner.
- [ ] New feature in a **backwards-compatible** manner.
- [x] Breaking change: bug fix or new feature that involve incompatible API changes.
- [ ] Other (e.g. doc update, configuration, etc)

### Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I added a line to the `CHANGELOG.md` file in the `Unreleased` section under the most fitting heading (e.g. `Added`, `Changed`, `Removed`).
- [x] I ran all tests on my computer and it's all green (i.e. `invoke test`).
- [x] I ran lint on my computer and there are no errors (i.e. `invoke lint`).
- [ ] I added new functions/classes and made them available on a second-level import, e.g. `compas_timber.datastructures.Beam`.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: src/compas_timber/assembly/assembly.py
Comment (obucklin) [inline src/compas_timber/assembly/assembly.py:1]: Yes serialize that asssembly data!

**Review by obucklin — APPROVED** [review]
Thanks Chen! I loked over the changes and mostly get what you are doing. You are right it is a good educaitonal opportunity. 

Maybe let me know if or when I should update my COMPAS or CT to use the new  serialization

File: requirements-dev.txt [hunk 1]
```diff
@@ -16,7 +16,7 @@ pydocstyle
 pytest >=3.2
 pytest-mock
 sphinx_compas_theme >=0.13
-sphinx >=3.4
+sphinx ==7.1.2
```
Comment (gonzalocasas) [inline requirements-dev.txt:19]: is this still required? higher version of sphinx break the theme?

**Review by gonzalocasas — APPROVED** [review]
lgtm! 

this got me thinking we should probably already bump the version in COMPAS, so that other dependencies such as this one can declare it properly, because now it's somewhat inconsistent, this version only works with `compas@main` but it's nowhere visible

File: requirements-dev.txt (hunk 1 above)
Comment (chenkasirer) [reply requirements-dev.txt:19]: yes `7.2.4` breaks the template, I opened https://github.com/compas-dev/sphinx_compas_theme/issues/20. 
Also Sphinx are seemingly dropping python versions like flies. `7.2.0` already doesn't support python 3.8 anymore..

**Comment by gonzalocasas** [comment]
> we can't add it to `requirements.txt` before there's an actual release of COMPAS 2.0. Is that what you meant?

yes, i don't know what's the best option for doing this, cut a 2.0-alpha release of COMPAS and use that?

**Comment by chenkasirer** [comment]
> > we can't add it to `requirements.txt` before there's an actual release of COMPAS 2.0. Is that what you meant?
> 
> yes, i don't know what's the best option for doing this, cut a 2.0-alpha release of COMPAS and use that?
> 
> 

I'd be all in for that. I feel like we shouldn't be too far from 2.0 anyways. @tomvanmele what do you say?

**Pull request merged**